### PR TITLE
Add notice about retiring Jaeger clients

### PR DESCRIPTION
**Pull request by yurishkuro**
Part of https://github.com/jaegertracing/jaeger/issues/3362

I am not sure how extensive the matrix of feature compatibility should be. In theory we could use the same format as in https://www.jaegertracing.io/docs/1.27/client-features/. It will involve more work to setup the rendering, but will result in easy to maintain source file and cleaner presentation.

### CODE REVIEW

**Comment by yurishkuro** [comment]
@jaegertracing/jaeger-maintainers  please review

File: Makefile [hunk 1]
```diff
@@ -4,8 +4,12 @@ THEME_DIR    := themes/$(HUGO_THEME)
...
 client-libs-docs:
 	@for d in $(shell ls -d content/docs/*); do \
-		cp content/_client_libs/*.md $$d/; \
-		echo "copied content/_client_libs/*.md -> $$d/"; \
+		pushd $$d/; \
+		rm -f client-libraries.md client-features.md; \
+		ln -s ../../_client_libs/client-libraries.md; \
+		ln -s ../../_client_libs/client-features.md; \
+		popd; \
+		echo "sym-linked content/_client_libs/*.md -> $$d/"; \
```
Comment (yurishkuro) [inline Makefile:12]: I decided to switch to symlinks because otherwise local editing of these files did not result in auto-rebuilding of the site

Comment (yurishkuro) [reply Makefile:12]: and, of course, it didn't work on netlify :-(

File: content/_client_libs/client-libraries.md [hunk 2]
```diff
@@ -7,9 +7,71 @@ children:
...
+
+For new applications, we recommend using the [OpenTelemetry](https://opentelemetry.io/) APIs and SDKs. For existing applications that are already instrumented with the OpenTracing API, we recommend replacing the Jaeger clients with the corresponding OpenTelemetry SDKs and the OpenTracing shim/bridge (available in most languages supported by Jaeger). We published a blog post that shows an example of the migration steps for Java: ["Migrating from Jaeger client to OpenTelemetry SDK"][blog-otel-java].
+
+### Timeline
+
+We plan to continue accepting pull requests and making new releases of Jaeger clients **through the end of 2021**. From 2022 we will enter a code freeze period **for 6 months**, during which we will no longer accept pull requests with new features, only with security-related fixes.  After that the client library repositories will be archived and accept no new changes.
```
Comment (albertteoh) [inline content/_client_libs/client-libraries.md:21]: I also suggest "accept no new changes" -> "stop accepting new changes"/"no longer accept new changes". Basically, I feel the negation should come first to make the intent clearer.

```suggestion
We plan to continue accepting pull requests and making new releases of Jaeger clients **through the end of 2021**. From 2022 we will enter a code freeze period **for 6 months**, during which we will no longer accept pull requests with new features, with the exception of security-related fixes.  After that the client library repositories will be archived and accept no new changes.
```

File: content/_client_libs/client-libraries.md [hunk 3]
```diff
@@ -7,9 +7,71 @@ children:
...
+
+### Migration to OpenTelemetry
+
+The OpenTelemetry project is working on publishing the migration guides from OpenTracing API to OpenTelemetry SDKs via OpenTracing bridges/shims. There may be different level of maturity and features in the SDKs. We will keep updating the information below as more of it becomes available.
```
Comment (albertteoh) [inline content/_client_libs/client-libraries.md:25]:
```suggestion
The OpenTelemetry project is working on publishing the migration guides from OpenTracing API to OpenTelemetry SDKs via OpenTracing bridges/shims. There may be different levels of maturity and features in the SDKs. We will keep updating the information below as more of it becomes available.
```

**Review by albertteoh — COMMENTED** [review]
> In theory we could use the same format as in https://www.jaegertracing.io/docs/1.27/client-features/.

+1

**Review by albertteoh — APPROVED** [review]
@yurishkuro would you like these changes to be part of the 1.28.0 release docs? If so, I can wait until this is merged.

**Comment by yurishkuro** [comment]
> would you like these changes to be part of the 1.28.0 release docs

these files are independent of version, so no dependency

**Pull request merged**